book/data_viz/plotting_data.nim

```diff
@@ -7,18 +7,521 @@ nbUseNimibook
+it *should just work* (few exceptions exist, e.g. `geom_raster` only draws fixed size tiles for performance).
+
+One final thing to mention: the `geom_histogram` procedures *also* take `data` and `aes`
+arguments. This means one can override the data or the aesthetics for a single geom!
+
+#### Applying geom layers to build the initial plot
+
+We now need to apply the input data and selection of columns to things we can actually draw.
+
+This is where the layering structure of `ggplot` actually becomes apparent, because from here
+we will list all kinds of `geoms` to draw. The order we list them directly determines
+the order they are drawn in.
+
+Step by step we will now add layer by layer and look at what happens with each to reproduce
+the plot talked about in the beginning. For that purpose we will generate a data frame that we
+will use in all following snippets. It will contain 3 columns:
+- "Energy": a column of twice 25 elements covering the range (0, 10) with 25 entries
+- "Counts": a column of twice 25 random entries between 0 and 10. The first 25 elements
+  are drawn as floats (to get fractional values) and the second 25 entries will be random
+  integer numbers
+- "Type": simply a column that designates the first 25 rows as "background" and the latter
+  25 as "calibration"
+
+Our construction in the following is a bit artificial of course.
+"""
+nbCode:
+  import ggplotnim, random, sequtils
+  randomize(42)
+  let df = seqsToDf({ "Energy" : cycle(linspace(0.0, 10.0, 25).toRawSeq, 2),
+                      "Counts" : concat(toSeq(0 ..< 25).mapIt(rand(10.0)),
+                                        toSeq(0 ..< 25).mapIt(rand(10).float)),
+                      "Type" : concat(newSeqWith(25, "background"),
+                                      newSeqWith(25, "candidates")) })
+  echo "Input data frame: "
+  echo "Head(10): ", df.head(10).pretty(10)
+  echo "Tail(10): ", df.tail(10).pretty(10)
 nbText: """
+What we want to achieve as a final plot is the following, where the explanations
+are mainly due to the original motivation of where the plot example is taken from:
+- a histogram for each "Type", drawn with one color each and a bit transparent so they
+  are visible where they overlap. Let each bin content correspond to some data point
+  at that energy.
+- also plot the actual data points on top of the bins. For the "background" like data
+  we have fractional values as it's values are normalized to the "candidates" (simply
+  by scaling from some hypothetical time for background / time for candidate data).
+- For the background data we want error bars. They represent our uncertainty of the
+  background hypothesis.
+- the candidates are just counts we measured. They don't have inherent uncertainties
+  (from a frequentist perspective we have to repeat the experiment many times and *then*
+  we can write down some variance in our candidates)
+
+##### Building layer 1 - `geom_histogram`
```

Comment:
This is just my opinion but at this point, the header is smaller than the text so I didn't even notice they were there at first. Even though it structurally might not be correct I would use `####` here instead for readability.

Reply:
Oh, thanks for catching that as well. I naively just used the next heading without thinking about that. Will do something about it. :)